lib/astutils.cpp

```diff
@@ -2173,17 +2173,28 @@ bool isVariableChanged(const Token *tok, int indirect, const Settings *settings,
         return !fun->isConst();
     }
 
-    const Token *ftok = tok2;
-    while (ftok && (!Token::Match(ftok, "[({]") || ftok->isCast()))
+    const Token* ftok = tok2, *ptok{};
+    while (ftok && (!Token::Match(ftok, "[({]") || ftok->isCast())) {
         ftok = ftok->astParent();
+        if (ftok && ftok->tokType() == Token::eLogicalOp)
+            ptok = ftok->astOperand1(); // check lhs
```

Comment:
Why are we moving to the lhs side of logical ops? Wont this miss the parent function call?

Reply:
The example looks something like `if (g(b) || b)`, and `g(b)` was not considered so far.

Reply:
Ah yea but you need to check `isVariableChanged` on the first `b` and not the second `b`. The first `b` should return true and the second `b` should return false.